lib/ddtrace/metrics.rb

```diff
@@ -122,6 +130,26 @@ def initialize(*args)
       end
     end
 
+    # Return instance specific tags.
+    def instance_tags
+      tags = []
+      tags << "#{Ext::Metrics::TAG_ENV}:#{env}" if env
+      tags
+    end
+
+    # Add instance tags to default metrics
+    def default_metric_options
+      # Return dupes, so that the constant isn't modified,
+      # and defaults are unfrozen for mutation in Statsd.
+      super.tap do |options|
+        options[:tags] = options[:tags].dup
+
+        # Add tags dynamically because they might change during
+        # runtime (i.e. when calling #configure).
+        options[:tags].concat(instance_tags)
+      end
+    end
```

Comment:
This logic is very similar to what we do in `Datadog::Runtime::Metrics`: https://github.com/DataDog/dd-trace-rb/blob/bcfeb95d21e3c0a2ded57388d6af0e23597215be/lib/ddtrace/runtime/metrics.rb#L69-L78